needle-slack-connector/src/server/slack/service.ts

```diff
@@ -0,0 +1,178 @@
+export interface TimezoneInfo {
+  timezone: string;
+  timezoneLabel: string;
+  timezoneOffset: number;
+}
+
+interface SlackUserResponse {
+  ok: boolean;
+  user?: {
+    tz: string;
+    tz_label: string;
+    tz_offset: number;
+  };
+}
+
+export interface SlackCanvasResponse {
+  ok: boolean;
+  files?: Array<{
+    id: string;
+    created: number;
+    timestamp: number;
+    name: string;
+    title: string;
+    mimetype: string;
+    filetype: string;
+    pretty_type: string;
+    user: string;
+    user_team: string;
+    editable: boolean;
+    size: number;
+    mode: string;
+    is_external: boolean;
+    external_type: string;
+    is_public: boolean;
+    public_url_shared: boolean;
+    display_as_bot: boolean;
+    username: string;
+    url_private: string;
+    url_private_download: string;
+    permalink: string;
+    url_static_preview: string;
+    quip_thread_id: string;
+    updated: number;
+    update_notification: number;
+    is_channel_space: boolean;
+    linked_channel_id: string;
+    channels: string[];
+    groups: string[];
+    ims: string[];
+    teams_shared_with: string[];
+    access: string;
+    comments_count: number;
+    title_blocks: Array<{
+      type: string;
+      block_id: string;
+      elements: Array<{
+        type: string;
+        elements: Array<{
+          type: string;
+          channel_id?: string;
+        }>;
+      }>;
+    }>;
+    last_read: number;
+    editors: string[];
+    edit_timestamp: number;
+    show_badge: boolean;
+  }>;
+  response_metadata?: {
+    next_cursor: string;
+  };
+  error?: string;
+}
+
+export function createSlackService(accessToken: string) {
```

Comment:
nit: javascript world i guess... closure instead of class 😄 nobody in JS uses classes - i actually like them.